### PR TITLE
DCIM-compliant camera patch (more info https://github.com/hrkdroid/Camera

### PR DESCRIPTION
**Pull request by hrk**
DCIM-compliant[1] camera patch (more info https://github.com/hrkdroid/CameraDCIM)

Bug has been reported to Android issue tracker [2] w/o an answer (and w/o a proper patch by me, forgive my newbie-ness, I'm waiting for official git tree to come back online for it).

Code is in use in an own-repackaged-app available on Android Market [3].

[1] http://en.wikipedia.org/wiki/Design_rule_for_Camera_File_system
[2] http://code.google.com/p/android/issues/detail?id=2960
[3] https://market.android.com/details?id=it.sineo.android.camera

### CODE REVIEW

File: src/com/android/camera/Camera.java [hunk 1]
```diff
@@ -742,7 +744,7 @@ public void onZoomChange(
         private int storeImage(byte[] data, Location loc) {
             try {
                 long dateTaken = System.currentTimeMillis();
-                String title = createName(dateTaken);
+                String title = /*createName(dateTaken)*/DCIMHelper.getNameForNewImage();
```
Comment (johnou) [inline src/com/android/camera/Camera.java:747]: No need to keep the old code in with a comment, version control is great :)

File: src/com/android/camera/ImageManager.java [hunk 2]
```diff
@@ -127,13 +128,25 @@ public int describeContents() {
...
+//    public static final String CAMERA_IMAGE_BUCKET_NAME =
+//            Environment.getExternalStorageDirectory().toString()
+//            + "/DCIM/Camera";
+//    public static final String CAMERA_IMAGE_BUCKET_ID =
+//            getBucketId(CAMERA_IMAGE_BUCKET_NAME);
+    /**
+     * DCIM standard compliant.
+     */
+    public static String CAMERA_IMAGE_BUCKET_NAME() {
+        return DCIMHelper.getCurrentDirectory();
```
Comment (johnou) [inline src/com/android/camera/ImageManager.java:140]: Call this directly from the caller?

**Comment by johnou** [comment]
Might have a better chance getting this pulled if you add the patch to gerrit => http://r.cyanogenmod.com/

**Comment by eljef** [comment]
Please submit through gerrit. http://review.cyanogenmod.org

**Pull request closed without merging**